data/devices/data-parse-test.py

```diff
@@ -190,6 +191,16 @@ def check_section_driver(driver, section):
         assert('Unsupported driver section {}'.format(driver))
 
 
+def validate_data_file_name(path):
+    # Matching any of the characters in the regular expression will throw an
+    # error. Currently only tests the square brackets [], parentheses, and curly
+    # braces.
+    illegal_characters_regex = '([\[\]\{\}\(\)])'
+    found_characters = re.findall(illegal_characters_regex, path)
+    if len(found_characters) > 0:
```

Comment:
I think you missed this one. Should be `if found_characters:`, python will evaluate 0 to False and >=1 to True.

Reply:
found_characters currently returns a list of matches. My goal was to spit out which characters were illegal later on in the request.

Reply:
Yes?

```python
if found_characters:
	raise AssertionError("data file name '{}' contains illegal characters: '{}'".format(path, ''.join(found_characters)))
```

An empty list will be evaluated to false (ignore my first comment about the 0 and >=1).

Reply:
Ah! I didn't know a test on an empty list would evaluate to False.

Reply:
Yes, python is pretty smart when evaluating things!